### PR TITLE
search

### PR DESCRIPTION
**Pull request by junho2000**
i think there is some problem at ADD
when i add third contact second contact removed and third contact go to second 

and also i revised contact, contactlist class due to delete reference type and add const keyword

### CODE REVIEW

File: contactlist.cpp [hunk 1]
```diff
@@ -10,8 +10,9 @@ void ContactList::add(Contact &contact) {
     this->_contacts.insert(contact.getPhoneNumber(), contact);
 
     // save contact to a file
-    QString filePath = QDir::currentPath() + "contact_data.bin";
-    QFile file(filePath);
+    //QString filePath = QDir::currentPath() + "contact_data.bin";
+
+    QFile file("contact_data.bin");
```
Comment (Shuta-syd) [inline contactlist.cpp:15]: this code also looks good, but you save a file like 15 line when the user opens the app and then saves a file, but this doesn't work.

That's why I think below the code is better.

```c++
    QString filePath = QDir::currentPath() + "contact_data.bin";
    QFile file(filePath);
```

File: mainwindow.cpp [hunk 2]
```diff
@@ -23,8 +23,7 @@ void MainWindow::on_btnAdd_clicked() { _addWin->exec(); }
 
 void MainWindow::on_btnSearch_clicked()
 {
-    searchWindow *searchWin = new searchWindow(this);
-
+    searchWindow *searchWin = new searchWindow(this, this->_phoneBook);
```
Comment (Shuta-syd) [inline mainwindow.cpp:26]: good, I think if mainWindow class has searchWindow class as a member valuable, you should initialize _searchWin in mainWindow constructor not like 26 line

**Review by Shuta-syd — COMMENTED** [review]
Thanks for your work!!! As a team we would be happy if you could write a comment in English!

File: contactlist.cpp (hunk 1 above)
Comment (Shuta-syd) [reply contactlist.cpp]: already resolved

File: mainwindow.cpp (hunk 2 above)
Comment (junho2000) [reply mainwindow.cpp:28]: haha okay!

**Review by Shuta-syd — APPROVED** [review]
LGTM

**Pull request merged**